### ISSUE TITLE
Add custom effect for lights

### ISSUE THREAD

**Issue by rmit-s3686930-keyu** [rospogrigio/localtuya#331]
In ``light.py`` I found
```
SCENE_LIST_RGBW_1000 = {
    "Night": "000e0d0000000000000000c80000",
    "Read": "010e0d0000000000000003e801f4",
    "Meeting": "020e0d0000000000000003e803e8",
    "Leasure": "030e0d0000000000000001f401f4",
    "Soft": "04464602007803e803e800000000464602007803e8000a00000000",
    "Rainbow": "05464601000003e803e800000000464601007803e803e80000000046460100f003e803"
    + "e800000000",
    "Shine": "06464601000003e803e800000000464601007803e803e80000000046460100f003e803e8"
    + "00000000",
    "Beautiful": "07464602000003e803e800000000464602007803e803e80000000046460200f003e8"
    + "03e800000000464602003d03e803e80000000046460200ae03e803e800000000464602011303e80"
    + "3e800000000",
}
```
After playing with it for a while I found:
```
"Shine": "06464601000003e803e800000000464601007803e803e80000000046460100f003e803e8"
    + "00000000",

A  B    C  D    E    F    G
06 4646 01 0000 03e0 03e8 00000000
```
A = No. of entry in the APP aka ``"scene_num"``
B = ``"unit_switch_duration" and "unit_gradient_duration"``
C = ``"unit_change_mode" [“static”,“jump”,“gradient”] (01-02)`` 
D = four bytes of hue ``(0-0168)``
E = four bytes of saturation ``(0-03e8)``
F = four bytes of brightness ``(0-03e8)``
G= padding (not sure)

I was able to manually add my on effect:
```
"Alert": "06606001016803e803e80000000060600100f003e803e800000000",
```
It's just flash red and blue

Can someone confirm if the data schema is correct and is there a way to add custom effect via GUI?

**Comment by rmit-s3686930-keyu** [comment]
Is there a tutorial on HA config-flow? I'm interested in trying to implement this feature myself

**Comment by KTibow** [comment]
See https://developers.home-assistant.io.

**Comment by brandonongzy** [comment]
I made a little Python script to decode or create new effect set. It is interactive and hope can be added to a GUI


[localtuyaDecoder.py.zip](https://github.com/rospogrigio/localtuya/files/6379643/localtuyaDecoder.py.zip)

**Comment by KTibow** [comment]
@brandonongzy Nice! Small suggestion: Format it with Black.

**Comment by brandonongzy** [comment]
> @brandonongzy Nice! Small suggestion: Format it with Black. Also consider uploading it as a GitHub gist/repo.

Thanks for the suggestion, I have published to a repo here [https://github.com/brandonongzy/LocalTuyaDecoder](https://github.com/brandonongzy/LocalTuyaDecoder) :)

**Comment by acs-lux** [comment]
Hi,

I finally figured out the dps for my led strip that didn't match the example. Now i can go crazy colours for party time locally.

Its great but my question is...

Is transition supported in localtuya? For me it has no affect on the speed of changes.

This is still worth all the effort since the official tuya is still cloud based and is the best thing i found for HA so far imho.

**Comment by KTibow** [comment]
@acs-lux I don't think that Tuya on its own supports transition, and localtuya doesn't add that. Effects (part of Tuya on its own) might be used to add that, though.

**Comment by acs-lux** [comment]
Ah ok thanks

**Comment by ThomasCr** [comment]
I will also be Interested on this, because I also miss some scenes.
https://github.com/rospogrigio/localtuya/issues/659

**Comment by Ayuus** [comment]
For those who like flashlights:

    "Flash Red":   "23646401016803e803e80000000060600100000000000000000000",
    "Flash Green": "23646401007803e803e80000000060600100000000000000000000",
    "Flash Blue":  "2364640100f003e803e80000000060600100000000000000000000",
    "Flash White": "246464010000000003e80000000060600100000000000000000000",
